pkg/generate/ack/controller.go

```diff
@@ -62,6 +62,16 @@ var (
 		"Dereference": func(s *string) string {
 			return *s
 		},
+		"AddToMap": func(m map[string]interface{}, k string, v interface{}) map[string]interface{} {
+			if len(m) == 0 {
+				m = make(map[string]interface{})
+			}
+			m[k] = v
+			return m
+		},
+		"Nil": func() interface{} {
+			return nil
+		},
```

Comment:
Interesting how this is returning an `interface{}` and plays well with `AddToMap`